### PR TITLE
Fixes, Round 4

### PR DESCRIPTION
**Pull request by mistergone**
This PR fixes two issues:
- When the URL offer contains an iped that is not found in the API, the page displays an error and allows no further interaction.
- Bar graphs properly display warnings when data is incomplete.

**Note**: There's probably some complaints to be made about the style of the page-level error displayed when an iped is invalid. I am happy to address them in a future PR, once folks have a chance to look and comment.
## Additions
- Added a catch for invalid ipeds which fires `financialView.badSchoolData()`, a new method
## Changes
- Fixed the display of bar graph notifications
## Testing
- Passes unit tests!
## Review
- @marteki @higs4281 - Pull it down, gulp, and check it out!
## Todos
- Fix styling of page-level error
## Checklist
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code can be automatically merged (no conflicts)
- [x] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
- [x] Passes all existing automated tests
- [x] New functions include new tests
- [x] New functions are documented (with a description, list of inputs, and expected output)
- [x] Placeholder code is flagged
- [x] Visually tested in supported browsers and devices
- [x] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

### CODE REVIEW

**Comment by marteki** [comment]
Incomplete data warnings are triggering for all programs I tested, and not triggering on programs that provided data.

The other additional items I could see for the form-level warning:
- Red outline around entire box, as specified at http://cfarm.github.io/design-manual/ui-toolkit/form-fields.html#form-level-alerts
- Second paragraph probably doesn't need to be bold.

👍  from me for merge. The style changes can be addressed with the remaining form level warnings we have to implement.

**Comment by higs4281** [comment]
👍 
- [x] missing or malformed iped raises error box
- [x] missing grad rate generates error box 

## Question

Shouldn't we raise a similar error for a bad/missing program ID or offer ID? We wouldn't want a student to complete a disclosure that didn't have a valid program attached, because it might not be accurate.  In the case of a bad offer ID, the notification will be void.
thoughts, @saintsoup52 ?

**Comment by marteki** [comment]
@higs4281 @saintsoup52 Yes, we will be doing the same thing for bad/missing program ID and offer ID. We have issues logged for those.

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,8 @@ All notable changes to this project will be documented in this file.
 We follow the [Semantic Versioning 2.0.0](http://semver.org/) format.
 
 ## Unreleased
+- Changed display of graph notifications for settlement schoolSalary
```
Comment (marteki) [inline CHANGELOG.md:5]: Hm. This PR deals with graduation rate and loan default rate, not the salary values. (I know because I'm working on them. :smile:)

**Comment by saintsoup52** [comment]
@higs4281 We definitely want to have error codes for improper PID and Offer IDs...

For demoing purposes  I still need the tool work without a PID or OID... I don't mind if there is a message that says the offer ID is missing, but the rest of the tool should work.

**Comment by mistergone** [comment]
Merging!

**Comment by higs4281** [comment]
@saintsoup52 

> For demoing purposes I still need the tool work without a PID or OID... I don't mind if there is a message that says the offer ID is missing, but the rest of the tool should work.

The bare `/offer/` page works on build.  
go there and try  
 `consumerfinance.gov/paying-for-college2/understanding-your-financial-aid-offer/offer/`

**Pull request merged**